packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -267,13 +262,7 @@ known, it can be explicitly provided to attach via the command-line, e.g.
       throwToolExit('Did not find any valid target devices.');
     }
 
-    final Artifacts? overrideArtifacts = device.artifactOverrides ?? _artifacts;
```

Comment:
`device.artifactOverrides` is no longer ever set as non-null, so this is no longer necessary

Reply:
I may be missing something here but it looks like on my end that `device.artifactOverrides` is still nullable on the `Device` class?

Reply:
Good catch, forgot to push a local commit